### PR TITLE
feat: add integration test for jinad

### PR DESCRIPTION
**Pull request by rutujasurve94**
- Integration test for `jinad`

- Download `jinaai/jinad` image from `Dockerhub`, install `jina`

- Use set of pre-embedded documents with for testing `indexing`

- Test the `query` (search) flow

- Uses `docker-compose` and `test-integration.sh`

• Use `test-pip` image and build it in the yaml.
• Install `jinad` requirements avoiding it installing jina again.
• Remove requirements installation from jina since it uses the `jina:test-pip` image.
• To try locally, please build `jina:test-pip` locally by running in the jina root:
`docker build --build-arg PIP_TAG="[devel]" -f Dockerfiles/pip.Dockerfile -t jinaai/jina:test-pip .`

### CODE REVIEW

**Comment by JoanFM** [comment]
I think it needs an extra folder level to hold every different test as a subfolder of `tests/integration/jinad/XX`

File: tests/integration/jinad/test_simple_distributed/Dockerfile
Comment (JoanFM) [inline tests/integration/jinad/test_simple_distributed/Dockerfile:21]: we do not need this for this test

File: tests/integration/jinad/test_simple_distributed/Dockerfile
Comment (JoanFM) [inline tests/integration/jinad/test_simple_distributed/Dockerfile:17]: or this

File: tests/integration/jinad/test_simple_distributed/test_integration.sh
Comment (JoanFM) [inline tests/integration/jinad/test_simple_distributed/test_integration.sh:28]: In this case the only thing we can expect is that we get a document in output with `text` equals to `hey, dude`

File: tests/integration/jinad/test_simple_distributed/test_integration.sh
Comment (JoanFM) [inline tests/integration/jinad/test_simple_distributed/test_integration.sh:19]: we do not need to update all these files

Comment (JoanFM) [reply tests/integration/jinad/test_simple_distributed/test_integration.sh:19]: just the yamlspec

File: .github/workflows/test-jinad.yml
Comment (JoanFM) [inline .github/workflows/test-jinad.yml:52]: Not for now, but I would have a `.sh` in the root of `jinad` folder that similar to pytest, captures every test in the child folders and executes them, and reports the failing ones.

File: tests/integration/jinad/test_simple_distributed/test_integration.sh
Comment (rutujasurve94) [reply tests/integration/jinad/test_simple_distributed/test_integration.sh:24]: done

File: .github/workflows/test-jinad.yml
Comment (JoanFM) [inline .github/workflows/test-jinad.yml:49]: rename this because it does not use `pytest`

File: .github/workflows/test-jinad.yml
Comment (rutujasurve94) [reply .github/workflows/test-jinad.yml:52]: This maps to the `jinad/jinad` folder (first directory `jinad` inside `jinad` repo)

File: tests/integration/jinad/test_simple_distributed/Dockerfile
Comment (JoanFM) [inline tests/integration/jinad/test_simple_distributed/Dockerfile:12]: this requirements may not be well addressed

**Review by JoanFM — COMMENTED** [review]
It looks like a good skeleton, I think we are missing now how to get it test what we want to test

File: tests/integration/jinad/test_simple_distributed/Dockerfile
Comment (rutujasurve94) [reply tests/integration/jinad/test_simple_distributed/Dockerfile]: updated the `Dockerfile` to the right path `tests/integration/test_distributed_simple/`

File: tests/integration/jinad/test_simple_distributed/docker-compose.yml
Comment (maximilianwerk) [inline tests/integration/jinad/test_simple_distributed/docker-compose.yml:4]: Remove the comment

File: tests/integration/jinad/test_simple_distributed/requirements.txt
Comment (maximilianwerk) [inline tests/integration/jinad/test_simple_distributed/requirements.txt]: I don't think we need these heavy libraries here for the simplified test.

File: tests/integration/jinad/test_simple_distributed/Dockerfile
Comment (maximilianwerk) [inline tests/integration/jinad/test_simple_distributed/Dockerfile:15]: I think we wanted to use the `jinad` image instead of installing `jinad` here. Or did this change?

File: tests/integration/jinad/test_simple_distributed/Dockerfile
Comment (maximilianwerk) [inline tests/integration/jinad/test_simple_distributed/Dockerfile:9]: I know this is copied from the other example, but I think the in-docker files can be less verbose. e.g. only do `COPY tests/integration/test_simple_distributed/requirements.txt /requirements.txt` and adapt the following `RUN` accordingly

File: .github/workflows/test-jinad.yml
Comment (maximilianwerk) [inline .github/workflows/test-jinad.yml:6]: For testing purposes, you can change this to 
```

on:
  pull_request:
#
```
and change it back to what is currently before merging. This helps you testing it with github actions, but without needing to merge it to master during this testing.

File: tests/integration/jinad/test_simple_distributed/docker-compose.yml
Comment (maximilianwerk) [inline tests/integration/jinad/test_simple_distributed/docker-compose.yml:21]: `JINAD_CONTEXT` is deprecated and has no meaning anymore.

**Review by maximilianwerk — CHANGES_REQUESTED** [review]
Well executed. Some minor adjustments and clean-ups.

**Comment by JoanFM** [comment]
We have to think about how to make `jinad` use the `jina-core` version being tested. But @nan-wang thought about having these tests on jinad to invert the dependency. What do you think @maximilianwerk @deepankarm @rutujasurve94 ?

**Comment by maximilianwerk** [comment]
> We have to think about how to make `jinad` use the `jina-core` version being tested. But @nan-wang thought about having these tests on jinad to invert the dependency. What do you think @maximilianwerk @deepankarm @rutujasurve94 ?

I am indifferent about this. You can also move this to `jinad` and watch there for merges to master in `jina`. Important thing: have some alarms ring somewhere, if we break it.

**Comment by JoanFM** [comment]
> > We have to think about how to make `jinad` use the `jina-core` version being tested. But @nan-wang thought about having these tests on jinad to invert the dependency. What do you think @maximilianwerk @deepankarm @rutujasurve94 ?
> 
> I am indifferent about this. You can also move this to `jinad` and watch there for merges to master in `jina`. Important thing: have some alarms ring somewhere, if we break it.

**Comment by deepankarm** [comment]
> We have to think about how to make `jinad` use the `jina-core` version being tested. But @nan-wang thought about having these tests on jinad to invert the dependency. What do you think @maximilianwerk @deepankarm @rutujasurve94 ?

We can have something simple to start with. **GET /alive** endpoint on remote can send `jina version` details to local. If there's a version mismatch, don't proceed. Once we have versioning for `jinad`, we can better plan this? https://github.com/jina-ai/jinad/issues/22

**Comment by JoanFM** [comment]
Let's have it working without blocking the PR from being merged and learn from there

**Comment by JoanFM** [comment]
image

This is happening now and I think is blocking the Request to be back to the caller

File: .github/workflows/test-jinad.yml
Comment (rutujasurve94) [reply .github/workflows/test-jinad.yml:6]: yes sure

File: tests/integration/jinad/test_simple_distributed/Dockerfile
Comment (rutujasurve94) [reply tests/integration/jinad/test_simple_distributed/Dockerfile:15]: Here, we are installing `jinad` with requirements, but using the `test-pip` version of jina for requirement, for consistency.

**Comment by JoanFM** [comment]
@rutujasurve94 can u reopen please? I closed by mistake

File: .github/workflows/test-jinad.yml
Comment (JoanFM) [inline .github/workflows/test-jinad.yml:38]: Avoid this test,

File: .github/workflows/test-jinad.yml
Comment (nan-wang) [inline .github/workflows/test-jinad.yml:41]:
```suggestion
          cd jina
          ./tests/integration/jinad/test_simple_distributed/test_integration.sh
```
I'm confused about this line. If this is the case, I would expect `test_integration.sh` under the `jinad` repo. However, we put it in the `jina` repo
```suggestion
          cd jinad/${{ matrix.path }}
          ./tests/integration/jinad/test_simple_distributed/test_integration.sh
```

File: tests/integration/jinad/test_simple_distributed/flow.yml
Comment (nan-wang) [inline tests/integration/jinad/test_simple_distributed/flow.yml:13]:
```suggestion
```

File: tests/integration/jinad/test_simple_distributed/flow.yml
Comment (nan-wang) [inline tests/integration/jinad/test_simple_distributed/flow.yml:19]:
```suggestion

```

**Review by nan-wang — DISMISSED** [review]

File: .github/workflows/test-jinad.yml
Comment (JoanFM) [reply .github/workflows/test-jinad.yml:41]: I would keep it like this now, and in another PR add a tests.sh in the jinad root an there put the logic to tesr every subfolder. 

And I woulf also kwep these tests in core repo to see every PR.

**Review by maximilianwerk — APPROVED** [review]

File: .github/workflows/test-jinad.yml
Comment (JoanFM) [reply .github/workflows/test-jinad.yml:41]: Ah well it is true that the real action has not been tested, I think @rutujasurve94 u can commit these suggesions

Comment (rutujasurve94) [reply .github/workflows/test-jinad.yml]: Done, doing `cd jina` instead of `matrix path` for jinad

**Review by JoanFM — DISMISSED** [review]

File: .github/workflows/test-jinad.yml
Comment (rutujasurve94) [reply .github/workflows/test-jinad.yml]: Sure, done.

**Review by JoanFM — APPROVED** [review]

**Review by hanxiao — APPROVED** [review]

**Pull request merged**
